### ISSUE TITLE
Tangram SegFaults on bad scene file from network or file 

### ISSUE THREAD

**Issue by polar** [tangrams/tangram-es#1864]
**TO REPRODUCE THE ISSUE, FOLLOW THESE STEPS:**

Create a bogus scene file 'badscene.html' that is NOT YAML, such as an error page you may get from a server:

```
<html>
  <head><title>Index of /</title></head>
  <body bgcolor="white">
   <h1>Index of /</h1><hr><pre><a href="../">../</a>
   <a href="directory/">directory/</a>                                           19-Jul-2018 13:56                   -
  </pre><hr></body>
  </html>
```
Then perform:

```
mapController.loadSceneFile("mapScenes/badscene.html");
```

Then you get:

```
08-29 11:15:45.338  7461  7461 D Tangram : TANGRAM map.cpp:230: Loading scene file: asset:///mapScenes/badscene.html
08-29 11:15:45.343  7461  7461 E libc++abi: terminating with uncaught exception of type YAML::BadSubscript: operator[] call on a scalar
08-29 11:15:45.343  7461  7461 F libc    : Fatal signal 6 (SIGABRT), code -6 in tid 7461 (.trimble.mileon), pid 7461 (.trimble.mileon)
08-29 11:15:45.421  7658  7658 I crash_dump64: obtaining output fd from tombstoned, type: kDebuggerdTombstone
```

**RESULT:**

The backtrace is attached.
[backtrace.txt](https://github.com/tangrams/tangram-es/files/2332700/backtrace.txt)

**EXPECTED RESULT:**

Describe what you expected to see here.
This error should not crash the app. 

**ENVIRONMENT:**

 - What operating system and device did you produce this issue on?
Android 8.1, Google Moto Project Fi Phone

 - If you used a released version, what is the version number?
Tangram-es:0.9.6 and 0.9.4

**OTHER:**

The code should catch this error and provide a SceneError back to the application.

If you are able to record the log output from tangram-es, please attach it as a text file or share it using a site like https://hastebin.com/

**Comment by polar** [comment]
This issue is solved by https://github.com/tangrams/tangram-es/pull/1865

**Comment by tallytalwar** [comment]
Addressed #1865, #1866
